kubernetes/nodecollection.go

```diff
@@ -53,12 +55,14 @@ func (cns ClientsetNodeSource) GetReadyNodes() ([]v1.Node, error) {
 
 	var readyNodes []v1.Node
 	for _, n := range allNodes.Items {
-		nodeReady := v1Node.IsNodeReady(&n)
-
-		if nodeReady {
+		i, nc := nodeutil.GetNodeCondition(
+			&n.Status,
+			v1.NodeReady)
+		log.Printf("status: %v %+v", i, nc)
+		if i >= 0 && nc.Type == v1.NodeReady {
```

Comment:
Do we need to check the `nc.Type==v1.NodeReady` if that is the type we are requesting in `GetNodeCondition`. I read the Godocs and it wasn't clear, so we can probably leave as is.